### PR TITLE
WIP Feat/poc seo ssr

### PR DESCRIPTION
**Pull request by achauve**
The PR is not in a state to be merged, it's a WIP of a POC to see if we can easily switch to an SSR stack to serve React profiles app, and in the meantime add static pages to replace wordpress and use it as a landing page.

Warning: a lot of things are broken (CSS / extension / ...).

### CODE REVIEW

File: src/api/sendInBlue/sendEmail.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { post } from './call';
-import { TransactionEmail } from 'SendInBlue';
+import { TransactionEmail } from './types';
```
Comment (lutangar) [inline src/api/sendInBlue/sendEmail.ts:2]: This is not supposed to be used by the profiles app.

File: src/api/sendInBlue/types.ts [hunk 2]
```diff
@@ -1,34 +1,32 @@
-declare module 'SendInBlue' {
-  export interface PersonEmail {
-    name?: string;
-    email: string;
-  }
-  export interface Sender extends PersonEmail {}
-  export interface Recipient extends PersonEmail {}
+export interface PersonEmail {
```
Comment (lutangar) [inline src/api/sendInBlue/types.ts:1]: Same, not supposed to be used by the profiles app.

File: readme_nextjs.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+The following suggested values were added to your tsconfig.json. These values can be changed to fit your project's needs:
```
Comment (lutangar) [inline readme_nextjs.md:1]: Might be remove before mergin, but can be usefull for the time being.

File: src/app/background/sagas/refreshContributors.ts [hunk 4]
```diff
@@ -8,6 +8,7 @@ export function* refreshContributorsPeriodicallySaga() {
     Number(process.env.REFRESH_CONTRIBUTORS_INTERVAL)
   );
 
+  // warning SSR: refresh LOOP
```
Comment (lutangar) [inline src/app/background/sagas/refreshContributors.ts:11]: Hum, `background` app is supposed to be the "heart" of the extension this shouldn't be included in the SSR process meaning:
- A user without the extension go to the profile app, all infos are supposed to come from the API not background.
- A user has the extension installed some of informations are fetched from background (ex: subscriptions)

File: src/app/background/sagas/refreshMatchingContexts.ts [hunk 5]
```diff
@@ -39,6 +39,7 @@ export function* refreshMatchingContextsPeriodicallySaga() {
     Number(process.env.REFRESH_MC_INTERVAL)
   );
 
+  // warning SSR: refresh LOOP
```
Comment (lutangar) [inline src/app/background/sagas/refreshMatchingContexts.ts:42]: See previous comment.

File: src/app/background/services/createContributionEmail.ts [hunk 6]
```diff
@@ -4,7 +4,7 @@ import {
   formatContributionTextEmail
 } from 'app/lmem/format/contribution';
 import { truncateWords } from '../../utils/truncate';
-import { TransactionEmail } from 'SendInBlue';
+import { TransactionEmail } from '../../../api/sendInBlue/types';
```
Comment (lutangar) [inline src/app/background/services/createContributionEmail.ts:7]: Same comment as mail related files that shouldn't be included in the profiles, thus not impacted by these changes.

File: src/app/profiles/App/Pages/Profiles/Profile/Profile.stories.tsx [hunk 7]
```diff
@@ -13,7 +12,7 @@ import {
 import { generateStatefulContributor } from 'test/fakers/generateContributor';
 import rootReducer from '../../../../store/reducers';
 
-const store = createStore(rootReducer(createBrowserHistory()));
+const store = createStore(rootReducer());
```
Comment (lutangar) [inline src/app/profiles/App/Pages/Profiles/Profile/Profile.stories.tsx:15]: There might be a way to ignore `*.stories.tsx` files from SSR.

File: src/app/profiles/App/Pages/index.tsx [hunk 8]
```diff
@@ -6,12 +6,13 @@ import Subscriptions from './Subscriptions';
 
 const Pages = () => (
   <>
-    <Switch>
+    <p>Hello pages index</p>
```
Comment (lutangar) [inline src/app/profiles/App/Pages/index.tsx:9]: I supposed routing would be handled by folder structure.

File: src/app/profiles/App/index.tsx [hunk 9]
```diff
@@ -1,26 +1,22 @@
...
-      <ConnectedRouter history={history}>
-        <Background>
-          <PageContainer>
-            <Pages />
-          </PageContainer>
-        </Background>
-      </ConnectedRouter>
-    </ThemeProvider>
-  </Provider>
+  <ThemeProvider theme={theme}>
```
Comment (lutangar) [inline src/app/profiles/App/index.tsx:11]: I wonder where the store is handled then...

File: src/app/profiles/store/sagas/index.ts [hunk 10]
```diff
@@ -8,13 +8,14 @@ import windowConnectionSaga from 'app/profiles/store/sagas/windowConnection.saga
...
-  : windowConnectionSaga.bind(null, process.env.PROFILES_ORIGIN);
+const connectionSaga =
+  typeof browser !== 'undefined' && browser?.runtime?.connect
+    ? portConnectionSaga
+    : windowConnectionSaga.bind(null, process.env.PROFILES_ORIGIN);
 
 export default function* rootSaga() {
   yield all([
-    fork(connectionSaga),
+    // fork(connectionSaga),
```
Comment (lutangar) [inline src/app/profiles/store/sagas/index.ts:18]: Yeah this part try to connect to the extension if it exists.

File: src/pages/aide.tsx [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+- Points communs et différences avec Twitter et Facebook
+
+## Vous ne trouvez pas la réponse à votre question ?
+
+Merci de nous contacter via le formulaire de contact
+ou bien par mail à support /_at_/ dismoi.io
+`;
+
+function Page() {
+  return <ContentPage title="Aide" content={pageText} path="/aide" />;
```
Comment (lutangar) [inline src/pages/aide.tsx:32]: :+1:

File: tsconfig.json [hunk 12]
```diff
@@ -11,7 +10,7 @@
     ] /* Specify library files to be included in the compilation. */,
     "allowJs": true /* Allow javascript files to be compiled. */,
     "checkJs": true /* Report errors in .js files. */,
-    "jsx": "react" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
+    "jsx": "preserve" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
```
Comment (lutangar) [inline tsconfig.json:13]: I wonder if it would be better to have a distinct config in profiles folder

**Comment by christpet** [comment]
Looks like this POC didn't have the results we wanted, i.e. we learned that it will actually be pretty complex and difficult to switch over to SSR. Is that correct @lutangar ? If yes let's close it!

**Comment by lutangar** [comment]
Indeed it'll require more work if we want to achieve this. Since @achauve won't work on this anymore and regarding the lack of time of other contributors, we must adapt our strategy.
So yes, I agree, it isn't a priority anymore, closing it.

**Pull request closed without merging**